File: server/src/main/java/de/zalando/zally/dto/SeverityBinder.java
package de.zalando.zally.dto;

import de.zalando.zally.rule.api.Severity;
import org.springframework.util.StringUtils;

import java.beans.PropertyEditorSupport;

public class SeverityBinder extends PropertyEditorSupport {

    @Override
    public void setAsText(final String text) throws IllegalArgumentException {
        final Severity value = StringUtils.hasText(text) ? Severity.valueOf(text.toUpperCase()) : null;
        setValue(value);
    }
}

File: server/zally-server/src/main/java/de/zalando/zally/dto/SeverityBinder.kt
package de.zalando.zally.dto

import de.zalando.zally.rule.api.Severity
import org.springframework.util.StringUtils

import java.beans.PropertyEditorSupport

class SeverityBinder : PropertyEditorSupport() {

    @Throws(IllegalArgumentException::class)
    override fun setAsText(text: String) {
        val value = if (StringUtils.hasText(text)) Severity.valueOf(text.toUpperCase()) else null
        setValue(value)
    }
}
